liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java

```diff
@@ -388,4 +390,33 @@ private String getPluginConfiguration(MavenProject proj, String pluginGroupId, S
         }
         return null;
     }
+    
+    /**
+     * Call {@link #updateArtifactPathToOutputDirectory(MavenProject,Artifact) updateArtifactPathToOutputDirectory(MavenProject mavenProject, Artifact artifactToUpdate)} 
+     * with <code>artifactToUpdate</code> obtained from <code>mavenProject</code>
+     * 
+     * @param mavenProject
+     */
+    protected void updateArtifactPathToOutputDirectory(MavenProject mavenProject) {
+    	updateArtifactPathToOutputDirectory(mavenProject, mavenProject.getArtifact());
+    }    
+    
+    /**
+     * Call <code>artifactToUpdate.setFile()</code> to build output directory (i.e. ".../target/classes") and
+     * also creates the directory if not present.  Together these will help avoid the core
+     * Maven artifact resolution trying to resolve the artifact against the local .m2, 
+     * fitting better into dev mode, "all-in-one" use cases.
+     * 
+     * @param mavenProject
+     * @param artifactToUpdate
+     */
+    protected void updateArtifactPathToOutputDirectory(MavenProject mavenProject, Artifact artifactToUpdate) {
+        File outputDir = Paths.get(mavenProject.getBuild().getOutputDirectory()).toFile();
+        artifactToUpdate.setFile(outputDir);
+        try {
+            outputDir.createNewFile();
```

Comment:
Any concern with creating an output directory for modules without any `src/main/java` code, ie. for ear modules? I can't think of any downside to doing this and just having the empty directory...